.github/workflows/release.yml

```diff
@@ -0,0 +1,35 @@
+name: CD Build and Publish to NPM
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      
+      - name: Yarn Install
+        run: yarn
+      
+      - name: Lint
+        run: yarn lint
+      
+      - name: Jest Tests
+        run: yarn test:ci
+      
+      - name: Typescript build
+        run: yarn build
+      
+      - name: Set Version to Release Tag Name
+        run: |
+          yarn version --new-version ${{ github.event.release.tag_name }} --no-git-tag-version
```

Comment:
I use the `--no-git-tag-version` as this will place a commit message onto master, this isn't needed as the git sha is captured in the release tagging process of github